pootle/apps/pootle_fs/utils.py

```diff
@@ -7,11 +7,76 @@
 # or later license. See the LICENSE file for a copy of the license and the
 # AUTHORS file for copyright and authorship information.
 
+from fnmatch import translate
+
+from django.utils.functional import cached_property
+
 from pootle.core.exceptions import MissingPluginError, NotConfiguredError
 
 from .delegate import fs_plugins
 
 
+class PathFilter(object):
+
+    def __init__(self, pootle_path=None, fs_path=None):
+        self.pootle_path = pootle_path
+        self.fs_path = fs_path
+
+    def filtered_pootle_path(self, qs):
+        if not self.pootle_regex:
+            return qs
+        return qs.filter(
+            pootle_path__regex=self.path_regex(self.pootle_path))
+
+    def filtered_fs_path(self, qs):
+        if not self.fs_regex:
+            return qs
+        return qs.filter(path__regex=self.fs_regex)
+
+    def filtered(self, qs):
```

Comment:
so qs is StoreFS queryset...
do we need `filtered_fs_path` and `filtered_pootle_path` separately from `filtered`?